src/index.js

```diff
@@ -143,6 +144,34 @@ export default function ({ template, types }) {
             }
           },
         });
+
+        programPath.scope.crawl();
+
+        programPath.traverse({
+          ImportDeclaration(path) {
+            if (!globalOptions.removeImport) {
+              return;
+            }
+            const { source, specifiers } = path.node;
+            if (source.value !== 'prop-types') {
+              return;
+            }
+
+            const usedSpecifiers = specifiers.reduce((usedSpecifiersArray, specifier) => {
+              const importedIdentifierName = specifier.local.name;
+              const { referencePaths } = path.scope.getBinding(importedIdentifierName);
+
+              if (referencePaths.length > 0) {
+                return [...usedSpecifiersArray, specifier];
+              }
+              return usedSpecifiersArray;
+            }, []);
+
+            if (usedSpecifiers.length === 0) {
+              path.remove();
```

Comment:
What about stopping the traverse once we find one occurrence?
https://github.com/thejameskyle/babel-handbook/blob/master/translations/en/plugin-handbook.md#stopping-traversal